### ISSUE TITLE
install_dell_bios_exe.py fails on XPS 13 9310 2-in-1 with latest firmware

### ISSUE THREAD

**Issue by Esokrates** [fwupd/fwupd#4788]
**Describe the bug**

Tried to install the latest bios update via the python script and the file I downloaded from the Dell website:

```
sudo /usr/share/fwupd/install_dell_bios_exe.py XPS_13_9310_2-in-1_2.10.0.exe 
Installing to System Firmware
Wrote capsule /tmp/tmp1i95f7bt/firmware.bin
GUID:      a472a89c-42ab-c171-228e-e182e424af76
HdrSz:     0x1000
Flags:     0x70000
PayloadSz: 0x1d61a70
Traceback (most recent call last):
  File "/usr/share/fwupd/install_dell_bios_exe.py", line 106, in <module>
    cab = generate_cab(ARGS.exe, directory, guid, version)
  File "/usr/share/fwupd/install_dell_bios_exe.py", line 57, in generate_cab
    create_firmware_cab(variables, directory)
  File "/usr/share/fwupd/firmware_packager.py", line 96, in create_firmware_cab
    subprocess.check_call(command)
  File "/usr/lib/python3.10/subprocess.py", line 364, in check_call
    retcode = call(*popenargs, **kwargs)
  File "/usr/lib/python3.10/subprocess.py", line 345, in call
    with Popen(*popenargs, **kwargs) as p:
  File "/usr/lib/python3.10/subprocess.py", line 966, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.10/subprocess.py", line 1842, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'gcab'
```
Rebooting did not start any upgrade.

**Steps to Reproduce**

1. Grab a 9310 2-in-1
2. Get the latest bios from https://www.dell.com/support/home/de-at/drivers/driversdetails?driverid=ttm64&oscode=wt64a&productcode=xps-13-9310-2-in-1-laptop
3. Fire up `sudo /usr/share/fwupd/install_dell_bios_exe.py XPS_13_9310_2-in-1_2.10.0.exe`

**Expected behavior**

The script should succeed and on next reboot the bios upgrade should start.

**fwupd version information**
Please provide the version of the daemon and client.

```shell
fwupdmgr --version
runtime   org.freedesktop.fwupd         1.7.5
compile   org.freedesktop.gusb          0.3.10
runtime   com.dell.libsmbios            2.4
runtime   org.kernel                    5.15.0-40-generic
compile   org.freedesktop.fwupd         1.7.5
runtime   org.freedesktop.gusb          0.3.10
```
latest version in Ubuntu 22.04 installed via apt.

Please note how you installed it (`apt`, `dnf`, `pacman`, source, etc):

**fwupd device information**
Please provide the output of the fwupd devices recognized in your system.

```shell
fwupdmgr get-devices --show-all-devices
XPS 13 9310 2-in-1
│
├─11th Gen Intel Core™ i7-1165G7 @ 2.80GHz:
│     Device ID:          4bde70ba4e39b28f9eab1628f9dd6e6244c03027
│     Current version:    0x000000a4
│     Vendor:             Intel
│     GUIDs:              b9a2dd81-159e-5537-a7db-e7101d164d3f
│                         30249f37-d140-5d3e-9319-186b1bd5cac3
│                         2347ad97-b545-5bae-b6b3-b70a9553cbba
│                         1df99df5-22f4-5505-98cc-22dcc53b132a
│     Device Flags:       • Internal device
│   
├─KBG40ZPZ1T02 NVMe KIOXIA 1024GB:
│     Device ID:          188dc728af57960ba9943289b8b32546baeac5f5
│     Summary:            NVM Express solid state drive
│     Current version:    10400105
│     Vendor:             KIOXIA Corporation (NVME:0x1E0F)
│     GUIDs:              fbef9958-2179-5fb1-9e91-31c22cc29b2d
│                         b2cca4ac-d65f-e811-9c2d-fa7ae01bbebc
│     Device Flags:       • Internal device
│                         • Updatable
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│   
├─System Firmware:
│ │   Device ID:          a45df35ac0e948ee180fe216a5f703f32dda163f
│ │   Summary:            UEFI ESRT device
│ │   Current version:    2.9.0
│ │   Minimum Version:    2.9.0
│ │   Vendor:             Dell (DMI:Dell Inc.)
│ │   Update State:       Success
│ │   GUIDs:              a472a89c-42ab-c171-228e-e182e424af76
│ │                       230c8b18-8d9b-53ec-838b-6cfc0383493a
│ │                       055d2140-2ac7-5568-ae08-d78b141978ae
│ │   Device Flags:       • Internal device
│ │                       • Updatable
│ │                       • System requires external power source
│ │                       • Supported on remote server
│ │                       • Needs a reboot after installation
│ │                       • Cryptographic hash verification is available
│ │                       • Device is usable for the duration of the update
│ │ 
│ └─UEFI dbx:
│       Device ID:        362301da643102b9f38477387e2193e57abaa590
│       Summary:          UEFI revocation database
│       Current version:  83
│       Minimum Version:  83
│       Vendor:           UEFI:Linux Foundation
│       Install Duration: 1 second
│       GUIDs:            00fe3755-a4d8-5ef7-ba5f-47979fbb3423
│                         4a6cd2cb-8741-5257-9d1f-89a275dacca7
│                         c6682ade-b5ec-57c4-b687-676351208742
│                         f8ba2887-9411-5c36-9cee-88995bb39731
│       Device Flags:     • Internal device
│                         • Updatable
│                         • Needs a reboot after installation
│                         • Only version upgrades are allowed
│     
├─TPM:
│     Device ID:          c6a80ac3a22083423992a3cb15018989f37834d6
│     Current version:    1.257.0.0
│     Vendor:             ST Microelectronics (TPM:STM)
│     GUIDs:              ff71992e-52f7-5eea-94ef-883e56e034c6
│                         84df3581-f896-54d2-bd1a-372602f04c32
│                         bfaed10a-bbc1-525b-a329-35da2f63e918
│                         70b7b833-7e1a-550a-a291-b94a12d0f319
│                         06f005e9-cb62-5d1a-82d9-13c534c53c48
│     Device Flags:       • Internal device
│   
├─TPM 2.0:
│     Device ID:          a3487e128cf1413519bce8e9a1ab3f5981e61458
│     Summary:            UEFI ESRT device
│     Current version:    0.1.1.1
│     Vendor:             Dell Inc. (PCI:0x1028)
│     Update State:       Success
│     Update Error:       Updating disabled due to TPM ownership
│     GUIDs:              f0bef8d6-125a-582e-97a4-2ebda39cff65
│                         ff71992e-52f7-5eea-94ef-883e56e034c6
│                         73730635-f6c2-53da-9df2-948bb5ac1022
│     Device Flags:       • Internal device
│                         • System requires external power source
│   
├─TigerLake-LP GT2 [Iris Xe Graphics]:
│     Device ID:          5792b48846ce271fab11c4a545f7a3df0d36e00a
│     Current version:    01
│     Vendor:             Intel Corporation (PCI:0x8086)
│     GUIDs:              c19276e0-d189-543b-88d6-67ccc0a46080
│                         99d36830-4701-5975-bf3d-57e06e1266c2
│                         3c11d563-4f83-5bd1-8369-75359ea6b6f6
│                         54ff3ed7-1ea3-5279-975f-93c13bf2aa53
│     Device Flags:       • Internal device
│                         • Cryptographic hash verification is available
│   
├─UEFI Device Firmware:
│     Device ID:          349bb341230b1a86e5effe7dfe4337e1590227bd
│     Summary:            UEFI ESRT device
│     Current version:    164
│     Minimum Version:    164
│     Vendor:             DMI:Dell Inc.
│     Update State:       Success
│     GUIDs:              176e090d-0ddb-495e-8173-bc998ccfecd0
│                         3e1d83c5-31af-55ac-9f88-8be02bda9596
│     Device Flags:       • Internal device
│                         • Updatable
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│   
├─UEFI Device Firmware:
│     Device ID:          2292ae5236790b47884e37cf162dcf23bfcd1c60
│     Summary:            UEFI ESRT device
│     Current version:    272630021
│     Minimum Version:    272630021
│     Vendor:             DMI:Dell Inc.
│     Update State:       Success
│     GUIDs:              b2cca4ac-d65f-e811-9c2d-fa7ae01bbebc
│                         0360499c-14d8-53b8-938b-37de40587af4
│     Device Flags:       • Internal device
│                         • Updatable
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│   
├─USB4 host controller:
│     Device ID:          e4deab7f7de6c24cd73a7a19b1fe671fb77fec6d
│     Summary:            Unmatched performance for high-speed I/O
│     Update Error:       No vendor ID set
│     GUID:               e72e778e-94f7-5ed2-b560-1c1262ee217c
│     Device Flags:       • Internal device
│                         • System requires external power source
│                         • Updatable
│   
├─USB4 host controller:
│     Device ID:          3a8249d0e4236bcb7fa19525259b563e04ba244a
│     Summary:            Unmatched performance for high-speed I/O
│     Update Error:       No vendor ID set
│     GUID:               e72e778e-94f7-5ed2-b560-1c1262ee217c
│     Device Flags:       • Internal device
│                         • System requires external power source
│                         • Updatable
│   
└─Wacom AES Device:
      Device ID:          dbbf40284cb602275c039924b932efc748233670
      Current version:    3006.00
      Vendor:             HIDRAW:0x056A
      GUIDs:              f2e31e0f-ab3b-5cd2-8a2f-462ada7ed50a
                          5f02cd21-7586-5650-95e9-388fdf0b6e63
                          a6531f36-4cd5-5878-a8f4-9efc1b7702b3
      Device Flags:       • Internal device
                          • Updatable
```

**System UEFI configuration**
Please provide the output of the following commands:

```shell
efibootmgr -v
BootCurrent: 0005
Timeout: 5 seconds
BootOrder: 0005,0001,0000,0007,0002,0004,0003
Boot0000* ubuntu	HD(1,GPT,95011be5-6358-4b24-b323-911bf76d0e26,0x800,0x100801)/File(\EFI\ubuntu\shimx64.efi)
Boot0001* Ubuntu 2110	PciRoot(0x0)/Pci(0x1d,0x0)/Pci(0x0,0x0)/NVMe(0x1,8C-E3-8E-04-00-FE-0E-43)/HD(1,GPT,8e6dbb0f-c812-4e7c-9cf3-83e558f44aec,0x800,0x113000)/File(\efi\ubuntu\shimx64.efi)
Boot0002* UEFI KBG40ZPZ1T02 NVMe KIOXIA 1024GB 808100FNN042 1	PciRoot(0x0)/Pci(0x1d,0x0)/Pci(0x0,0x0)/NVMe(0x1,8C-E3-8E-04-00-FE-0E-43)/HD(1,GPT,8e6dbb0f-c812-4e7c-9cf3-83e558f44aec,0x800,0x113000)/File(\EFI\Boot\BootX64.efi)N.....YM....R,Y.
Boot0003* UEFI Samsung Type-C 0376321020002841	PciRoot(0x0)/Pci(0xd,0x0)/USB(1,0)/HD(1,GPT,8bd89f6a-68cb-448a-b90a-77fda7d0e283,0x800,0x100801)/File(\EFI\Boot\BootX64.efi)N.....YM....R,Y.
Boot0004* Linux Firmware Updater	HD(1,GPT,8e6dbb0f-c812-4e7c-9cf3-83e558f44aec,0x800,0x113000)/File(\EFI\ubuntu\fwupdx64.efi)
Boot0005* ubuntu	HD(1,GPT,8bd89f6a-68cb-448a-b90a-77fda7d0e283,0x800,0x100801)/File(\EFI\ubuntu\shimx64.efi)
Boot0007* Windows Boot Manager	HD(1,GPT,8e6dbb0f-c812-4e7c-9cf3-83e558f44aec,0x800,0x113000)/File(\EFI\Microsoft\Boot\bootmgfw.efi)WINDOWS.........x...B.C.D.O.B.J.E.C.T.=.{.9.d.e.a.8.6.2.c.-.5.c.d.d.-.4.e.7.0.-.a.c.c.1.-.f.3.2.b.3.4.4.d.4.7.9.5.}...3................
```

```shell
efivar -l | grep fw
0abba7dc-e516-4167-bbf5-4d9d1c739416-fwupd-a472a89c-42ab-c171-228e-e182e424af76-0
0abba7dc-e516-4167-bbf5-4d9d1c739416-fwupd-ux-capsule
```

```shell
tree /boot
/boot
├── config-5.15.0-40-generic
├── efi
│   └── EFI
│       ├── BOOT
│       │   ├── BOOTX64.EFI
│       │   ├── fbx64.efi
│       │   └── mmx64.efi
│       └── ubuntu
│           ├── BOOTX64.CSV
│           ├── grub.cfg
│           ├── grubx64.efi
│           ├── mmx64.efi
│           └── shimx64.efi
├── grub
│   ├── fonts
│   │   └── unicode.pf2
│   ├── grub.cfg
│   ├── grubenv
│   ├── locale
│   │   ├── de.mo
│   │   ├── en_AU.mo
│   │   ├── en_CA.mo
│   │   ├── en_GB.mo
│   │   └── en@quot.mo
│   ├── unicode.pf2
│   └── x86_64-efi
│       ├── acpi.mod
│       ├── adler32.mod
│       ├── affs.mod
│       ├── afs.mod
│       ├── afsplitter.mod
│       ├── ahci.mod
│       ├── all_video.mod
│       ├── aout.mod
│       ├── appleldr.mod
│       ├── archelp.mod
│       ├── ata.mod
│       ├── at_keyboard.mod
│       ├── backtrace.mod
│       ├── bfs.mod
│       ├── bitmap.mod
│       ├── bitmap_scale.mod
│       ├── blocklist.mod
│       ├── boot.mod
│       ├── bsd.mod
│       ├── bswap_test.mod
│       ├── btrfs.mod
│       ├── bufio.mod
│       ├── cat.mod
│       ├── cbfs.mod
│       ├── cbls.mod
│       ├── cbmemc.mod
│       ├── cbtable.mod
│       ├── cbtime.mod
│       ├── chain.mod
│       ├── cmdline_cat_test.mod
│       ├── cmp.mod
│       ├── cmp_test.mod
│       ├── command.lst
│       ├── configfile.mod
│       ├── core.efi
│       ├── cpio_be.mod
│       ├── cpio.mod
│       ├── cpuid.mod
│       ├── crc64.mod
│       ├── cryptodisk.mod
│       ├── crypto.lst
│       ├── crypto.mod
│       ├── cs5536.mod
│       ├── ctz_test.mod
│       ├── datehook.mod
│       ├── date.mod
│       ├── datetime.mod
│       ├── diskfilter.mod
│       ├── disk.mod
│       ├── div.mod
│       ├── div_test.mod
│       ├── dm_nv.mod
│       ├── echo.mod
│       ├── efifwsetup.mod
│       ├── efi_gop.mod
│       ├── efinet.mod
│       ├── efi_uga.mod
│       ├── ehci.mod
│       ├── elf.mod
│       ├── eval.mod
│       ├── exfat.mod
│       ├── exfctest.mod
│       ├── ext2.mod
│       ├── extcmd.mod
│       ├── f2fs.mod
│       ├── fat.mod
│       ├── file.mod
│       ├── fixvideo.mod
│       ├── font.mod
│       ├── fshelp.mod
│       ├── fs.lst
│       ├── functional_test.mod
│       ├── gcry_arcfour.mod
│       ├── gcry_blowfish.mod
│       ├── gcry_camellia.mod
│       ├── gcry_cast5.mod
│       ├── gcry_crc.mod
│       ├── gcry_des.mod
│       ├── gcry_dsa.mod
│       ├── gcry_idea.mod
│       ├── gcry_md4.mod
│       ├── gcry_md5.mod
│       ├── gcry_rfc2268.mod
│       ├── gcry_rijndael.mod
│       ├── gcry_rmd160.mod
│       ├── gcry_rsa.mod
│       ├── gcry_seed.mod
│       ├── gcry_serpent.mod
│       ├── gcry_sha1.mod
│       ├── gcry_sha256.mod
│       ├── gcry_sha512.mod
│       ├── gcry_tiger.mod
│       ├── gcry_twofish.mod
│       ├── gcry_whirlpool.mod
│       ├── geli.mod
│       ├── gettext.mod
│       ├── gfxmenu.mod
│       ├── gfxterm_background.mod
│       ├── gfxterm_menu.mod
│       ├── gfxterm.mod
│       ├── gptsync.mod
│       ├── grub.efi
│       ├── gzio.mod
│       ├── halt.mod
│       ├── hashsum.mod
│       ├── hdparm.mod
│       ├── hello.mod
│       ├── help.mod
│       ├── hexdump.mod
│       ├── hfs.mod
│       ├── hfspluscomp.mod
│       ├── hfsplus.mod
│       ├── http.mod
│       ├── iorw.mod
│       ├── iso9660.mod
│       ├── jfs.mod
│       ├── jpeg.mod
│       ├── json.mod
│       ├── keylayouts.mod
│       ├── keystatus.mod
│       ├── ldm.mod
│       ├── legacycfg.mod
│       ├── legacy_password_test.mod
│       ├── linux16.mod
│       ├── linuxefi.mod
│       ├── linux.mod
│       ├── loadbios.mod
│       ├── load.cfg
│       ├── loadenv.mod
│       ├── loopback.mod
│       ├── lsacpi.mod
│       ├── lsefimmap.mod
│       ├── lsefi.mod
│       ├── lsefisystab.mod
│       ├── lsmmap.mod
│       ├── ls.mod
│       ├── lspci.mod
│       ├── lssal.mod
│       ├── luks2.mod
│       ├── luks.mod
│       ├── lvm.mod
│       ├── lzopio.mod
│       ├── macbless.mod
│       ├── macho.mod
│       ├── mdraid09_be.mod
│       ├── mdraid09.mod
│       ├── mdraid1x.mod
│       ├── memdisk.mod
│       ├── memrw.mod
│       ├── minicmd.mod
│       ├── minix2_be.mod
│       ├── minix2.mod
│       ├── minix3_be.mod
│       ├── minix3.mod
│       ├── minix_be.mod
│       ├── minix.mod
│       ├── mmap.mod
│       ├── moddep.lst
│       ├── modinfo.sh
│       ├── morse.mod
│       ├── mpi.mod
│       ├── msdospart.mod
│       ├── mul_test.mod
│       ├── multiboot2.mod
│       ├── multiboot.mod
│       ├── nativedisk.mod
│       ├── net.mod
│       ├── newc.mod
│       ├── nilfs2.mod
│       ├── normal.mod
│       ├── ntfscomp.mod
│       ├── ntfs.mod
│       ├── odc.mod
│       ├── offsetio.mod
│       ├── ohci.mod
│       ├── part_acorn.mod
│       ├── part_amiga.mod
│       ├── part_apple.mod
│       ├── part_bsd.mod
│       ├── part_dfly.mod
│       ├── part_dvh.mod
│       ├── part_gpt.mod
│       ├── partmap.lst
│       ├── part_msdos.mod
│       ├── part_plan.mod
│       ├── part_sun.mod
│       ├── part_sunpc.mod
│       ├── parttool.lst
│       ├── parttool.mod
│       ├── password.mod
│       ├── password_pbkdf2.mod
│       ├── pata.mod
│       ├── pbkdf2.mod
│       ├── pbkdf2_test.mod
│       ├── pcidump.mod
│       ├── pgp.mod
│       ├── play.mod
│       ├── png.mod
│       ├── priority_queue.mod
│       ├── probe.mod
│       ├── procfs.mod
│       ├── progress.mod
│       ├── raid5rec.mod
│       ├── raid6rec.mod
│       ├── random.mod
│       ├── rdmsr.mod
│       ├── read.mod
│       ├── reboot.mod
│       ├── regexp.mod
│       ├── reiserfs.mod
│       ├── relocator.mod
│       ├── romfs.mod
│       ├── scsi.mod
│       ├── search_fs_file.mod
│       ├── search_fs_uuid.mod
│       ├── search_label.mod
│       ├── search.mod
│       ├── serial.mod
│       ├── setjmp.mod
│       ├── setjmp_test.mod
│       ├── setpci.mod
│       ├── sfs.mod
│       ├── shift_test.mod
│       ├── signature_test.mod
│       ├── sleep.mod
│       ├── sleep_test.mod
│       ├── smbios.mod
│       ├── spkmodem.mod
│       ├── squash4.mod
│       ├── strtoull_test.mod
│       ├── syslinuxcfg.mod
│       ├── tar.mod
│       ├── terminal.lst
│       ├── terminal.mod
│       ├── terminfo.mod
│       ├── test_blockarg.mod
│       ├── testload.mod
│       ├── test.mod
│       ├── testspeed.mod
│       ├── tftp.mod
│       ├── tga.mod
│       ├── time.mod
│       ├── tpm.mod
│       ├── trig.mod
│       ├── tr.mod
│       ├── true.mod
│       ├── udf.mod
│       ├── ufs1_be.mod
│       ├── ufs1.mod
│       ├── ufs2.mod
│       ├── uhci.mod
│       ├── usb_keyboard.mod
│       ├── usb.mod
│       ├── usbms.mod
│       ├── usbserial_common.mod
│       ├── usbserial_ftdi.mod
│       ├── usbserial_pl2303.mod
│       ├── usbserial_usbdebug.mod
│       ├── usbtest.mod
│       ├── video_bochs.mod
│       ├── video_cirrus.mod
│       ├── video_colors.mod
│       ├── video_fb.mod
│       ├── videoinfo.mod
│       ├── video.lst
│       ├── video.mod
│       ├── videotest_checksum.mod
│       ├── videotest.mod
│       ├── wrmsr.mod
│       ├── xfs.mod
│       ├── xnu.mod
│       ├── xnu_uuid.mod
│       ├── xnu_uuid_test.mod
│       ├── xzio.mod
│       ├── zfscrypt.mod
│       ├── zfsinfo.mod
│       ├── zfs.mod
│       └── zstd.mod
├── initrd.img -> initrd.img-5.15.0-40-generic
├── initrd.img-5.15.0-40-generic
├── initrd.img.old -> initrd.img-5.15.0-40-generic
├── memtest86+.bin
├── memtest86+.elf
├── memtest86+_multiboot.bin
├── System.map-5.15.0-40-generic
├── vmlinuz -> vmlinuz-5.15.0-40-generic
├── vmlinuz-5.15.0-40-generic
└── vmlinuz.old -> vmlinuz-5.15.0-40-generic

8 directories, 306 files
```

**Additional questions**

- Operating system and version: Ubuntu 22.04
- Have you tried rebooting? Yes
- Is this a regression? Yes, since it was fixed in the past: https://github.com/fwupd/fwupd/issues/2840
- Are you using an NVMe disk? Yes
- Is secure boot enabled? Yes
- Is this a Lenovo system with 'Boot Order Lock' turned on in the BIOS? No

@paulmenzel Do you observe this as well?

**Comment by hughsie** [comment]
> No such file or directory: 'gcab'

You need to install the `gcab` CLI program.

**Comment by CragW** [comment]
After you have `gcab` installed in your system, you'd probably face another issue in association with LVFS signing check. The fix 148e2b83 has landed in 1.8.0 to extend the support of installing Dell's system firmware anyway.
